lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -0,0 +1,391 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""## Overview.
+
+This document explains how to use the `CertHandler` class to
+create and manage TLS certificates through the `tls_certificates` interface.
+
+The goal of the CertHandler is to provide a wrapper to the `tls_certificates`
+library functions to make the charm integration smoother.
+
+## Library Usage
+
+This library should be used to create a `CertHandler` object, as per the
+following example:
+
+```python
+self.cert_handler = CertHandler(
+    charm=self,
+    key="my-app-cert-manager",
+    cert_subject="unit_name",  # Optional
+)
+```
+
+You can then observe the library's custom event and make use of the key and cert:
+```python
+self.framework.observe(self.cert_handler.on.cert_changed, self._on_server_cert_changed)
+
+container.push(keypath, self.cert_handler.key)
+container.push(certpath, self.cert_handler.cert)
+```
+
+This library requires a peer relation to be declared in the requirer's metadata. Peer relation data
```

Comment:
this can go

Reply:
and maybe we should add a notice that this library can't be used if your juju version doesn't support secrets?